packages/shared/components/molecules/FoundryDetails.svelte

```diff
@@ -0,0 +1,96 @@
+
+    $: formattedTransactionTime = formatDate(transactionTime, { dateStyle: 'long', timeStyle: 'medium' })
+    $: hasStorageDeposit = storageDeposit || (storageDeposit === 0 && giftedStorageDeposit === 0)
+
+    $: formattedStorageDeposit = formatTokenAmountPrecise(
+        storageDeposit ?? 0,
+        BASE_TOKEN[$activeProfile?.networkProtocol]
+    )
+
+    $: formattedGiftedStorageDeposit = formatTokenAmountPrecise(
+        giftedStorageDeposit ?? 0,
+        BASE_TOKEN[$activeProfile?.networkProtocol]
+    )
+
+    $: localePrefix = `tooltips.transactionDetails.${direction === ActivityDirection.In ? 'incoming' : 'outgoing'}.`
+
+    let detailsList: { [key in string]: { data: string; tooltipText?: string } }
+    $: detailsList = {
+        transactionTime: { data: formattedTransactionTime },
+        ...(hasStorageDeposit && {
+            storageDeposit: {
+                data: formattedStorageDeposit,
+                tooltipText: localize(localePrefix + 'storageDeposit'),
+            },
+        }),
+        ...(giftedStorageDeposit && {
+            giftedStorageDeposit: {
+                data: formattedGiftedStorageDeposit,
+                tooltipText: localize(localePrefix + 'giftedStorageDeposit'),
+            },
+        }),
+    }
+</script>
+
+<transaction-details class="w-full h-full space-y-6 flex flex-auto flex-col flex-shrink-0">
+    <main-content class="flex flex-auto w-full flex-col items-center justify-center space-y-3">
+        {#if amount}
+            <transaction-value class="flex flex-col items-center">
+                <div class="flex flex-row space-x-3">
+                    <AssetIcon {asset} />
+                    <div class="flex flex-row flex-wrap justify-center items-baseline space-x-0.1">
+                        <Text type={TextType.h1} fontWeight={FontWeight.semibold}>{amount}</Text>
+                        {#if unit}
+                            <Text type={TextType.h4} classes="ml-1" fontWeight={FontWeight.medium}>{unit}</Text>
+                        {/if}
+                    </div>
+                </div>
+                {#if formattedFiatValue}
+                    <Text fontSize="md" color="gray-600" darkColor="gray-500">{formattedFiatValue}</Text>
+                {/if}
+            </transaction-value>
+        {/if}
+        {#if subject?.type === 'account'}
+            <Box row clearBackground clearPadding classes="justify-center">
+                <AccountLabel account={subject.account} />
+            </Box>
+        {:else if subject?.type === 'address'}
+            <AddressBox clearBackground clearPadding isCopyable address={subject?.address} />
+        {:else}
+            <Box row clearBackground clearPadding classes="justify-center">
+                <Text type={TextType.pre} fontSize="base" fontWeight={FontWeight.medium}>
+                    {localize('general.unknownAddress')}
+                </Text>
+            </Box>
+        {/if}
```

Comment:
This will be used across multiple activity details so we can extract into its own component